workflow/scripts/tmb.py

```diff
@@ -59,107 +60,118 @@
-    VEP_INFO = INFO.split("CSQ=")[1]
-    Variant_type = VEP_INFO.split("|")[1].split("&")
-    db1000G = VEP_INFO.split("|")[41]
-    if db1000G == "":
-        db1000G = 0
-    else:
-        db1000G = float(db1000G)
-    GnomAD = VEP_INFO.split("|")[47]
-    if GnomAD == "":
-        GnomAD = 0
-    else:
-        GnomAD = float(GnomAD)
-    db = VEP_INFO.split("|")[17]
-    FORMAT = lline[8].split(":")
-    AD_index = 0
-    DP_index = 0
-    VD_index = 0
-    i = 0
-    for f in FORMAT:
-        if f == "AD":
-            AD_index = i
-        elif f == "DP":
-            DP_index = i
-        elif f == "VD":
-            VD_index = i
-        i += 1
-    DATA = lline[9].split(":")
-    AD = DATA[AD_index].split(",")
-    if len(AD) == 2:
-        VD = int(AD[1])
-    else:
-        VD = int(DATA[VD_index])
-    DP = int(DATA[DP_index])
-
-    # Artifact observations
-    Observations = 0
-    if len(ref) == 1 and len(alt) == 1:
-        if key in FFPE_SNV_artifacts:
-            Observations = FFPE_SNV_artifacts[key]
-    else:
-        continue
+vep_dict = {}
+with gzip.open(vcf, 'rt') as vcf_infile:
+    file_content = vcf_infile.read().split("\n")
+    for line in file_content:
+        if line == "":
+            continue
+        if header:
+            if line[:6] == "#CHROM":
+                header = False
+            if line[:14] == "##INFO=<ID=CSQ":
+                vep_string = line.split("Format: ")[1]
+                vep_list = vep_string.split("|")
+                i = 0
+                for v in vep_list:
+                    vep_dict[v] = i
+                    i += 1
+            continue
+        lline = line.strip().split("\t")
+        chrom = lline[0]
+        pos = lline[1]
+        if chrom == prev_chrom and pos == prev_pos:
+            continue
+        prev_pos = pos
+        prev_chrom = chrom
+        key = chrom + "_" + pos
+        ref = lline[3]
+        alt = lline[4]
+        filter = lline[6]
+        INFO = lline[7]
+        if INFO[:3] == "AA=":
+            continue
+        INFO_list = INFO.split(";")
+        AF_index = 0
+        Caller_index = 0
+        i = 0
+        for info in INFO_list:
+            if info[:3] == "AF=":
+                AF_index = i
+            if info[:8] == "CALLERS=":
+                Caller_index = i
+            i += 1
+        AF = float(INFO_list[AF_index][3:])
+        Callers = INFO_list[Caller_index]
+        nr_Callers = len(Callers.split(","))
+        if nr_Callers < 2:
+            continue
+        VEP_INFO = INFO.split("CSQ=")[1].split("|")
+        Variant_type = VEP_INFO[vep_dict["Consequence"]].split("&")
+        db1000G = VEP_INFO[vep_dict["AF"]]
+        if db1000G == "":
+            db1000G = 0
+        else:
+            db1000G = float(db1000G)
+        GnomAD = VEP_INFO[vep_dict["MAX_AF"]]
+        if GnomAD == "":
+            GnomAD = 0
+        else:
+            GnomAD = float(GnomAD)
+        FORMAT = lline[8].split(":")
+        AD_index = 0
```

Comment:
```suggestion
        format_index = {field: index for index, field in enumerate(FORMAT)}
```
should make it easier to add field in later steps in the script, but not required (just a suggestion). If applied we more changes are needed ex: DP = int(DATA[format_index['DP']])

Reply:
skippar detta nu men bra tips